src/parsers/manifest/dash/node_parsers/utils.ts

```diff
@@ -129,11 +275,77 @@ function parseScheme(root: Element) : IScheme {
            value };
 }
 
+/**
+ * Create a function to factorize the MPD parsing logic.
+ * @param {Object} dest - The destination object which will contain the parsed
+ * values.
+ * @param {Array.<Error>} warnings - An array which will contain every parsing
+ * error encountered.
+ * @return {Function}
+ */
+function ValueParser<T>(
+  dest : T,
+  warnings : Error[]
+) {
+  /**
+   * Parse a single value and add it to the `dest` objects.
+   * If an error arised while parsing, add it at the end of the `warnings` array.
+   * @param {string} objKey - The key which will be added to the `dest` object.
+   * @param {string} val - The value found in the MPD which we should parse.
+   * @param {Function} parsingFn - The parsing function adapted for this value.
+   * @param {string} displayName - The name of the key as it appears in the MPD.
+   * This is used only in error formatting,
+   */
+  return function(
+    objKey : keyof T,
+    val : string,
+    parsingFn : (val : string,
+                 displayName : string) => [T[keyof T] | null, MPDError | null],
+    displayName : string
+  ) : void {
+    const [parsingResult, parsingError] = parsingFn(val, displayName);
+    if (parsingError !== null) {
+      log.warn(parsingError.message);
```

Comment:
Seems a little bit strange to warn here about the parsing error.

Reply:
Where should I warn though?